### ISSUE TITLE
checksum verification failed for vertex/module-tax

### ISSUE THREAD

**Issue by rommelfreddy** [magento/magento2#19934]
### Preconditions (*)
1. Magento CE 2.2.3

### Steps to reproduce (*)
1. Upgrade with composer (guided by devdocs)

### Expected result (*)
1. upgraded magento

### Actual result (*)
1. The checksum verification of the file failed for module vertex/module-tax

### Comment 
The same issue happended with a other module. (https://github.com/magento/magento2/issues/16129)
I have tried the same things but I don't get it work.
Does anybody has problems with this packaged too ?

**Comment by magento-engcom-team** [comment]
Hi @Fredwak. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@Fredwak do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [x] no

**Comment by CajuCLC** [comment]
Having same issue. On Magento Commerce 2.2.2, tried to update using composer and getting error:

```
  - Installing vertex/sdk (1.0.0): Loading from cache
  - Installing vertex/module-tax (3.0.0): Downloading (100%)         
                                                                                                                                                    
  [UnexpectedValueException]                                                                                                                        
  The checksum verification of the file failed (downloaded from https://repo.magento.com/archives/vertex/module-tax/vertex-module-tax-3.0.0.0.zip)
```

composer.json updated:

"magento/product-enterprise-edition": "2.3",
mcom to latest version.
module-avatax to latest version.
Added the zend\\mvc\\controller.
Ran composer update and the vertex error happens.

I tried deleting vendor/vertex, but same error.

**Comment by magento-engcom-team** [comment]
Hi @milindsingh. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by vladsh91** [comment]
the same problem. i can reproduce this bug with the following commands:

```
mv ~/.cache/composer ~/.cache/composer-old
composer create-project --repository=https://repo.magento.com/ magento/project-community-edition magento2
```

**Comment by Tom-T** [comment]
I'm having the same issue installing magneto using both create project as well as using a composer.json file with this module in the remove section.

**Comment by alex4allofyou** [comment]
Yeah, have it too.

`  [UnexpectedValueException]
  The checksum verification of the file failed (downloaded from https://repo.
  magento.com/archives/vertex/module-tax/vertex-module-tax-3.0.0.0.zip)`

And when I did replace, later during install of Magento2 says it cannot find it and aborts install, even though I did not select it.

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@dmytro-ch Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97305` were created

**Comment by qqil** [comment]
Same error:
```

composer create-project --repository=https://repo.magento.com/ magento/project-community-edition ./magento


Installing magento/project-community-edition (2.3.0)
  - Installing magento/project-community-edition (2.3.0): Downloading (100%)
Created project in ./magento
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 392 installs, 0 updates, 0 removals
  - Installing magento/magento-composer-installer (0.1.13): Downloading (100%)
  - Installing symfony/polyfill-ctype (v1.10.0): Downloading (100%)
...............
  - Installing vertex/sdk (1.0.0): Downloading (100%)
  - Installing vertex/module-tax (3.0.0): Downloading (100%)
In FileDownloader.php line 178:

  The checksum verification of the file failed (downloaded from https://repo.magento.com/archives/vertex/module-tax/vertex-module-tax-3.0.0.0.zip)

```

**Comment by milindsingh** [comment]
I think it's similar  #19283

**Comment by DanielRuf** [comment]
It's a duplicate issue imo. I think we can close the other issue.

**Comment by edwinljacobs** [comment]
Having the same problem here.

**Comment by Stepa4man** [comment]
Hey, what the status of this issue?
Are there any workarounds?
I have the same issue for 2.3.0, installing project with a composer from scratch

**Comment by DanielRuf** [comment]
See https://github.com/magento/magento2/issues/19283#issuecomment-442472271 an the other possible solutions.

**Comment by Stepa4man** [comment]
Hey, @DanielRuf, thanks, but it doesn't work for me:
![image](https://user-images.githubusercontent.com/15912461/50398797-11aa6280-0783-11e9-8d1d-8ea44e1e5044.png)
I'm trying to install new project.

**Comment by andrewbess** [comment]
Hey, everybody
I also have this problem.
How to fix it?

**Comment by DanielRuf** [comment]
> Hey, everybody
> I also have this problem.
> How to fix it?

See the linked issue. It's basically an issue with the composer lock file.
And there are also some solutions posted like getting the checksum from the archive and updating the lockfile.

**Comment by DanielRuf** [comment]
> Hey, everybody
> I also have this problem.
> How to fix it?

See the linked issue. It's basically an issue with the composer lockfile.
And there are also some solutions posted like getting the checksum from the archive and updating the lockfile.

**Comment by andrewbess** [comment]
> > Hey, everybody
> > I also have this problem.
> > How to fix it?
> 
> See the linked issue. It's basically an issue with the composer lockfile.
> And there are also some solutions posted like getting the checksum from the archive and updating the lockfile.

Sorry, but I need to install (not update) Magento using composer according to devdoc instructions https://devdocs.magento.com/guides/v2.3/install-gde/composer.html
So I cannot fix lockfile. It isn't exists before installation.

**Comment by DanielRuf** [comment]
You probably have to clear the composer cache where its metadata is still stored.

**Comment by andrewbess** [comment]
> You probably have to clear the composer cache where its metadata is still stored.

I have already done it.
@DanielRuf , please try to create new project using command
composer create-project --repository=https://repo.magento.com/ magento/project-community-edition
to reproduce it.

**Comment by antoinekociuba** [comment]
I confirm that this is not a composer caching issue, because it happens straight away on a fresh new box, without any composer cache on it.

**Comment by alex4allofyou** [comment]
Composer caching? Done it multiple times. It is because of this useless extension which most do not need. It is a shame that these things are not optional, shame that they reside within the core installation. Most of them I do not need / cannot use here in Europe anyway.

**Comment by mflasquin** [comment]
> Composer caching? Done it multiple times. It is because of this useless extension which most do not need. It is a shame that these things are not optional, shame that they reside within the core installation. Most of them I do not need / cannot use here in Europe anyway.

This is why the only solution for the moment is the following https://magento.stackexchange.com/questions/255537/the-checksum-verification-of-the-file-failed-downloaded-from-https-repo-magen :)

**Comment by rommelfreddy** [comment]
@all:
there is a little workaround for the issue with THIS package:
simply add the following to your composer.json and delete the lock file:

`"replace": {
        "vertex/module-tax": "*"
    },`
But you should valid if you do not need the package in your project.
A other workaround is that you download the package and provide it as archive in your composer.json.
(There are similar issues - also in other frameworks - just google it and you will figure it out how it works - I have never done it before)

**Comment by cmuench** [comment]
That is a problem on the Magento side.

The composer.json in the Composer cache expects that for Magento 2.3.0:

    "dist": {
          "type": "zip",
          "url": "https://repo.magento.com/archives/vertex/module-tax/vertex-module-tax-3.0.0.0.zip",
          "reference": null,
          "shasum": "5301f7a4d294cabcab3ae83e31fe5386c076520c"
        },

I downloaded the "vertex-module-tax-3.0.0.0.zip" manually.
The shasum is `dd5e23680878e70fdff546fa661822b100ab2bf3` which is different. That's why composer stops the download.

**Comment by metalagman** [comment]
Having the same issue with checksum. I believe the `vertex/module-tax` was published with a wrong checksum.

**Comment by ravmenon** [comment]
@Fredwak @cmuench @russianlagman There was an issue setting up vertex/module-tax 3.0.0 on the repo server side.  The issue is fixed now and you should now see the correct checksum dd5e23680878e70fdff546fa661822b100ab2bf3. 

A fresh install of 2.3.0 should work now.  

Any existing installs may have to clear the composer cache and/or update the composer.lock file to fix the stale shasum with the above value.

**Comment by cmuench** [comment]
@ravbetsky Thanks for solving. I can confirm that the current installation works also for Magento Commerce 2.3.0 and Commerce 2.2.7.

**Comment by antoinekociuba** [comment]
Thx for the reactivity on this issue guys, I can also confirm that it's OK now.

**Comment by fanny00** [comment]
hi, 
I get this error when trying to install composer for magento 2.3

[UnexpectedValueException]
  The checksum verification of the file failed (downloaded from https://repo.
  magento.com/archives/vertex/module-tax/vertex-module-tax-3.0.0.0.zip)

thanks!

**Comment by janmyszkier** [comment]
@DanielRuf bumping this topic because this is still an issue. 
vertex/module-tax is still having checksum issues today for 
```
"require": {
        "magento/product-community-edition": "2.3.0"
    },
```

composer.lock shows:
```
{
            "name": "vertex/module-tax",
            "version": "3.0.0",
            "dist": {
                "type": "zip",
                "url": "https://repo.magento.com/archives/vertex/module-tax/vertex-module-tax-3.0.0.0.zip",
                "reference": null,
                "shasum": "5301f7a4d294cabcab3ae83e31fe5386c076520c"
            },
```
but it seems to NOT be it (says composer for all of us)
Also, please note the referenced topic (19283) is about different package (vertex/sdk) and not vertex/module-tax

Can magento stop releasing /accepting fixes under old release numbers? 

if that was a bugfix can you please just release it as a new number following semver and eventually making specific (broken) release unavailable? Because this looks like what happened with the vertex/module-tax here.

Currently the only way to fix this is modify composer.lock checksum from `5301f7a4d294cabcab3ae83e31fe5386c076520c` to `dd5e23680878e70fdff546fa661822b100ab2bf3` after every `composer update`

**Comment by craigcarnell** [comment]
Having the same issue today

**Comment by emastyle** [comment]
Seems same issue today...
  [UnexpectedValueException]
  The checksum verification of the file failed (downloaded from https://repo.magento.com/archives/vertex/module-tax/vertex-module-tax-3.0.0.0.zip)

**Comment by LiamKarlMitchell** [comment]
Still a problem.
